cli/src/pcluster/templates/slurm_builder.py

```diff
@@ -328,12 +332,18 @@ def _add_slurm_compute_fleet(self):
         # terminate_compute_fleet_custom_resource.add_depends_on()
 
     def _add_private_hosted_zone(self):
-        cluster_hosted_zone = route53.CfnHostedZone(
-            self.stack_scope,
-            "Route53HostedZone",
-            name=self.cluster_dns_domain.value_as_string,
-            vpcs=[route53.CfnHostedZone.VPCProperty(vpc_id=self.config.vpc_id, vpc_region=self._stack_region)],
-        )
+        if self._condition_custom_cluster_dns():
+            subdomain = self.stack_name
+            hosted_zone_id = self.config.scheduling.settings.dns.hosted_zone_id
+            domain_name = AWSApi.instance().route53.get_domain_name(hosted_zone_id)
+            cluster_hosted_zone = CustomDns(ref=hosted_zone_id, name=subdomain + domain_name)
```

Comment:
Let's double check the domain name coming from route53 contains the `.` we need to add between cluster name and domain_name.

Reply:
Forgot to update the change with "." to remote. Updated